All_home_works/hw8/task2.py

```diff
@@ -0,0 +1,58 @@
+"""
+Description in task2.rst
+"""
+import sqlite3
+
+
+class TableData:
+    def __init__(self, db_name, db_table):
+        self._db_name = db_name
+        self._db_table = db_table
+
+    def check_if_table_in_db(self):
+        self.cursor.execute(
+            "SELECT name FROM sqlite_master WHERE type='table' AND name=?;",
+            (self._db_table,),
+        )
+        if self.cursor.fetchone() is None:
+            raise sqlite3.OperationalError(f"no such table: {self._db_table}")
+
+    def __enter__(self):
+        self._conn = sqlite3.connect(self._db_name)
+        self._conn.row_factory = sqlite3.Row
+        self.cursor = self._conn.cursor()
+        self.check_if_table_in_db()
+        return self
+
+    def __iter__(self):
+        self.cursor.execute(f"SELECT * FROM '{self._db_table}';")  # noqa: S608
+        return self
```

Comment:
Let's change this code a bit.   Instead of returning self let's return the iterator here.  After that, we can remove `__next__`